tests/unit/s2n_random_test.c

```diff
@@ -839,10 +839,9 @@ static int s2n_random_invalid_urandom_fd_cb(struct random_test_case *test_case)
         EXPECT_OK(s2n_rand_set_urandom_for_test());
 
         EXPECT_TRUE(dev_urandom->fd > STDERR_FILENO);
-        if (test == 0) {
-            /* Close the file descriptor. */
-            EXPECT_EQUAL(close(dev_urandom->fd), 0);
-        } else {
+        /* Close the file descriptor. */
+        EXPECT_EQUAL(close(dev_urandom->fd), 0);
+        if (test != 0) {
             /* Make the file descriptor invalid by pointing it to STDERR. */
             dev_urandom->fd = STDERR_FILENO;
         }
```

Comment:
As discussed offline, you're making the difference between case i==0 and i==1 less clear. i==0 is specifically trying to test a closed fd. i==1 is specifically trying to test an invalid fd, and you only close the fd to avoid a leak.

I would suggest some combination of adding a comment explaining what the test is testing, keeping the if/else, and adding a comment about why each case is closing the fd.